### ISSUE TITLE
Fan in/out appears to stop once new commit lands

### ISSUE THREAD

**Issue by plainsane** [gocd/gocd#4776]
##### Issue Type
- Bug Report
- Documentation Report
##### Summary
When building a pipeline graph by specifying upstream pipelines as materials, works well until a new commit lands on the repo.  Once this happens, the older invactions of the pipeline no longer trigger their down streams.
##### Environment
aws ec2 instance
###### Basic environment details
amazon linux 1 ami
openjdk 8
script-executor plugin
yaml plugin
##### Steps to Reproduce
fork git@github.com:plainsane/gogogadgetcd.git
add as config repo to server xml.
wait for pipeline A to reach its first manual approval stage.
make new commit to repo and push.
wait for pipeline A to reach its first manual approval stage.
select history on pipeline A
allow instance 1 of pipeline A to continue to completion. 
##### Expected Results
i would expect pipeline B and C to trigger for the same sha as pipeline A's shaw.
##### Actual Results
nothing happens at all.  if you allow the latest instance of pipeline A to complete, things behave as expected.

##### Any other info
i have a pipeline graph of 13 pipelines that im using to control the flow of code/terraform from dev->qa->production.  i currenlty have to lock out all developers from landing code on my master branch to take code to qa, then to production (which can take half a day sometimes).  i would really like to allow things to progress from a specific sha while code is landed on master in a fail forward fashion but i currently can not do that as pipelines fail to trigger downstream once a new commit lands on master.

**Comment by varshavaradarajan** [comment]
Could reproduce this with above config - The go server logs show 

```
BuildCauseProducerService:175 - Error while scheduling pipeline: CR3. Possible Reasons: (1) Upstream pipelines have not been built yet. (2) Materials do not match between configuration and build-cause.
2018-05-24 13:39:12,981 WARN  [91@MessageListener for ScheduleCheckListener] BuildCauseProducerService:175 - Error while scheduling pipeline: CR2. Possible Reasons: (1) Upstream pipelines have not been built yet. (2) Materials do not match between configuration and build-cause.
```

Need to debug further.

**Comment by plainsane** [comment]
so the fact that pipeline A was built with a specific sha that has been superseded by a new sha in the repository, i can no longer progress pipeline B or C for the sha that pipeline A was built against?

i was under the impression that pipeline B and C would be schedule for the sha that pipeline A completed for stage apply.  i didnt see anytrhing in the documentation that described this scenario.

**Comment by varshavaradarajan** [comment]
@plainsane - it does when I had a similar config in the xml file instead of a config repo. So, either its a setup issue or a bug. Needs to be debugged further.

**Comment by plainsane** [comment]
thanx for the quick responses.  my main reason for filing this is to answer this question.
if i have a long chain of pipelines that all share the same scm material, and i get half way through my 10 pipelines triggering each other.  should a new commit to the scm stop the rest of the chaining and force me to start over, going back through the previously completed pipelines?  we have ppl that continue to land code after a deploment begins.

im sorry for going back over this, its a very difficult scenario for me to communicate correctly.

**Comment by ibnc** [comment]
I'm looking into a fix for this issue

**Comment by ibnc** [comment]
This issue is happening because the sha for the first run of pipeline A doesn't match the sha for the configuration changes.

From what I see in the commit history, this was done intentionally to prevent builds from running with out of date or invalid configuration. 

None of the above is the actual bug here, from what I can see anyway. The actual bug here is that non configuration changing commits are updating the configuration's revision when there is no need to do so.

I'm not working next week, so someone else is gonna need to pick this up

**Comment by plainsane** [comment]
"None of the above is the actual bug here, from what I can see anyway. The actual bug here is that non configuration changing commits are updating the configuration's revision when there is no need to do so."  and i would expect any change in configuration to abort the pipelines.  i really appreciate everyone's help on this issue.  thanx a million.

**Comment by ibnc** [comment]
I'm back from vacation. I'll start looking into a solution for this issue @plainsane

**Comment by ibnc** [comment]
@plainsane A fix for this issues has been merged, and should be released in 18.7

Thank you for your patience!

**Comment by ibnc** [comment]
@plainsane Could you download the latest experimental release, and confirm that these changes fix the issue? 

You can download it at https://www.gocd.org/download/ on the experimental tab

**Comment by rajiesh** [comment]
Verified on 18.7.0 (7117-1709149e6946f9cec95e4a8e914a47c7806f0672). @plainsane  - Let know if the fix did not address your issue
